### DISCUSSION TITLE
TUI alternative for Notion-like tables/databases

### DISCUSSION THREAD

**Discussion by jennydaman** [Feel-ix-343/markdown-oxide#170] [Ideas]
Hello,

I am a fellow Notion/Anytype refugee, looking to limit my exposure to proprietary, electron, and non-rust apps...

Does anyone here know of a TUI alternative to Notion-like databases? Being compatible with Obsidian, it would be great for markdown-oxide to have some analog to the [dataview](https://github.com/blacksmithgu/obsidian-dataview) and/or [obsidian-db-folder](https://rafaelgb.github.io/obsidian-db-folder/) Obsidian plugins.

If not, this is something I might want to start developing. I am imagining a software like this: given a directory of markdown files, render each markdown file's YAML front matter (a.k.a. Obsidian _metadata_) as table rows. The software would watch for filesystem changes and re-render when a file is deleted, created, or changed.

**Comment by Feel-ix-343** [comment]
Oxide version 1 will have advanced querying with filters and (likely) vector search.

this could be written to a file as a code block too

we may be able to do something like this
